libc/src/setjmp/arm/longjmp.cpp

```diff
@@ -0,0 +1,26 @@
+
+//===-- Implementation of longjmp -----------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "src/setjmp/longjmp.h"
+#include "src/__support/common.h"
+
+namespace LIBC_NAMESPACE {
+
+[[gnu::naked]]
+LLVM_LIBC_FUNCTION(void, longjmp, (__jmp_buf * buf, int val)) {
+  asm(R"(
+      ldm r0, {r4-r12, lr}
```

Comment:
Is it possible to have (link to) documentations / comments to help newbies like me on these asm?

Reply:
The definitive document on ARM assembler (32b and 64b) is called the Arm ARM.

https://developer.arm.com/documentation/ddi0487/latest/ is the stable link to the latest edition of the non-stable document.

AAPCS is also handy: https://github.com/ARM-software/abi-aa?tab=readme-ov-file#document-locations

https://nickdesaulniers.github.io/blog/2023/03/10/disambiguating-arm/

Reply:
This code is restoring r4-r11, sp, and lr from the memory referred to by `buf`.

It then returns `1` if `val` was zero, else `val`.  Since `lr` was restored, it doesn't return to the caller of `longjmp`, but instead the caller of `setjmp`.

Reply:
If you'd like a quick primer on ARM assembler, such as the calling convention, let's hop on a video call?

Reply:
added comments PTAL